A78-grpc-metrics-wrr-pf-xds.md

```diff
@@ -0,0 +1,148 @@
+A78: gRPC OTel Metrics for WRR, Pick First, and XdsClient
```

Comment:
I have a high level question about this design: existing metrics, such as those defined in A66, are implemented using the call tracer plugin mechanism, called out in A66, such as `StreamTracer` in java and `stats.Handler` in Go. Those allow co-existence of opencensus and opentelemetry, an example that is given in A66. I don't see it specified in this gRFC to have callbacks for each one of those new metrics rather than direct use of opentelemetry instrumentation. And I am asking because we make heavy use of a statsd based implementation internally, and ideally those new instrumentations are not gated behind a migration to opentelemetry. We also have Datadog customers that might prefer continuing with their statsd based approach rather than moving to OpenTelemetry for their existing services, and still make use of those new metrics.

Otherwise those metrics that are not per-call are going to be very useful, thank you for adding them!

Reply:
@yashykt is working on another gRFC that will be a companion to this one, which will spell out the framework for supporting non-per-call metrics.  That other gRFC is currently reffered to in this doc as "A??"; that link will be updated once the PR for that gRFC is created and the number assigned.  These two gRFCs will move forward in lock step, since they are part of the same effort; the only reason for splitting this up into two gRFCs is that some people are interested in the metrics we're defining but not in the underlying architecture.

I think the right place for this discussion will be that other gRFC, which will be created soon.  However, as a quick high-level answer to your question, the framework we're building will indeed provide a layer of abstraction between the stats plugin (e.g., OpenCensus or OpenTelemetry) and the metrics exported from within gRPC.  However, at least initially, the API to implement new stats plugins will not be public, and the only implementation we will support will use OpenTelemetry.  But as the framework matures, we may decide to expose it as a public API.

Reply:
OK, thank you. Sounds like this other gRFC is what I'll want to have a closer look at.

Feel free to resolve.